cognitive/src/test/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPromptSuite.scala

```diff
@@ -149,6 +257,100 @@ class OpenAIPromptSuite extends TransformerFuzzing[OpenAIPrompt] with OpenAIAPIK
       .count(r => Option(r.getSeq[String](0)).isDefined)
   }
 
+  test("getResponseFormat should return the default response format") {
+    val prompt = new OpenAIPrompt()
+    prompt.getResponseFormat shouldEqual ""
+  }
+
+  test("setResponseFormat should set the response format correctly with String") {
+    val prompt = new OpenAIPrompt()
+    prompt.setResponseFormat("json")
+    prompt.getResponseFormat shouldEqual "json_object"
+
+    prompt.setResponseFormat("json_object")
+    prompt.getResponseFormat shouldEqual "json_object"
+
+    prompt.setResponseFormat("text")
+    prompt.getResponseFormat shouldEqual "text"
+  }
+
+
+  test("setResponseFormat should throw an exception for invalid response format") {
+    val prompt = new OpenAIPrompt()
+    an[IllegalArgumentException] should be thrownBy {
+      prompt.setResponseFormat("invalid_format")
+    }
+  }
+
+  test("setResponseFormat should set the response format correctly with ResponseFormat") {
+    val prompt = new OpenAIPrompt()
+    prompt.setResponseFormat(OpenAIResponseFormat.JSON)
+    prompt.getResponseFormat shouldEqual "json_object"
+
+    prompt.setResponseFormat(OpenAIResponseFormat.TEXT)
+    prompt.getResponseFormat shouldEqual "text"
+  }
+
+
+  test("setResponseFormat should set the response format correctly for valid values") {
+    val prompt = new OpenAIPrompt()
+    prompt.setResponseFormat("text")
+    prompt.getResponseFormat should be ("text")
+
+    prompt.setResponseFormat("json")
+    prompt.getResponseFormat should be ("json_object")
+
+    prompt.setResponseFormat("json_object")
+    prompt.getResponseFormat should be ("json_object")
+
+    prompt.setResponseFormat("jSoN")
+    prompt.getResponseFormat should be ("json_object")
+
+    prompt.setResponseFormat("TEXT")
+    prompt.getResponseFormat should be ("text")
+  }
```

Comment:
these tests look duplicated from above, if the core parameter flexibility is shared between the two we can keep just 1 copy and still have same basic coverage

Reply:
removed